agent/src/main/java/org/openjdk/jmc/agent/TransformRegistry.java

```diff
@@ -83,8 +85,10 @@
 	 * @param xmlDescription
 	 *            an XML snippet describing the wanted modifications.
 	 * @return a set of class names associated with modified {@link TransformDescriptor}s.
+	 * @throws XMLStreamException
+	 *             if the supplied XML fails to validate.
 	 */
-	Set<String> modify(String xmlDescription);
+	Set<String> modify(String xmlDescription) throws XMLStreamException;
```

Comment:
Perhaps have our own exception class (e.g. XmlValidationException)? It shouldn't depend on whatever XML parser methology we're using (StAX vs DOM etc).

Reply:
Makes sense, I've added an XMLValidationException in the most recent commit and have it thrown from validateProbeDefinition